Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -6915,6 +6915,55 @@ static NSTextAlignment nsTextAlignmentFromTextAlignment(TextAlignment textAlignm
 
 #endif // ENABLE(IMAGE_ANALYSIS_ENHANCEMENTS)
 
+#if ENABLE(CONTENT_INSET_BACKGROUND_FILL)
+
+void WebViewImpl::updateFixedContentExtensionViews()
+{
+    if (!page().protectedPreferences()->contentInsetBackgroundFillEnabled())
+        return;
+
+    RetainPtr webView = view();
+    auto& fixedContainerEdges = [webView _coreFixedContainerEdges];
+    auto insets = obscuredContentInsets();
+    auto bounds = [webView bounds];
+
+    auto updateExtensionView = [&](BoxSide side, Function<NSRect()>&& computeFrame) {
+        bool needsView = insets.at(side) > 0 && fixedContainerEdges.fixedEdges.at(side);
+        RetainPtr extensionView = m_fixedColorExtensionViews.at(side);
+        if (needsView) {
+            if (!extensionView) {
+                extensionView = adoptNS([NSView new]);
+                [extensionView setWantsLayer:YES];
+                [extensionView layer].name = @"Fixed color extension fill";
```

Comment:
I wonder if it will help if you also included the box side.

Reply:
Oh, good idea! Will add the side.